### PR TITLE
Read manifest and blobs from bucket

### PR DESCRIPTION
**Pull request by unmultimedio**
Logic used in push command to go from a bucket to manifest and files blob is also useful in the backend. Moving it to the `manifest` pkg.

### CODE REVIEW

File: private/pkg/manifest/manifest.go [hunk 1]
```diff
@@ -194,6 +194,15 @@ func (m *Manifest) UnmarshalText(text []byte) error {
...
 
+// Blob returns the manifest's blob.
+func (m *Manifest) Blob() (Blob, error) {
+	manifestText, err := m.MarshalText()
+	if err != nil {
+		return nil, err
+	}
+	return NewMemoryBlobFromReader(bytes.NewReader(manifestText))
+}
+
```
Comment (twilly) [inline private/pkg/manifest/manifest.go:205]: 👏

File: private/pkg/manifest/module.go [hunk 2]
```diff
@@ -166,6 +187,15 @@ func (s *BlobSet) BlobFor(digest string) (Blob, bool) {
 	return blob, true
 }
 
+// Blobs returns the blobs in the set as an array.
+func (s *BlobSet) Blobs() []Blob {
+	var blobs []Blob
```
Comment (twilly) [inline private/pkg/manifest/module.go:192]:
```suggestion
	blobs := make([]Blob, len(s.digestToBlob))
```
... should reduce memory allocations.

File: private/pkg/manifest/module.go [hunk 3]
```diff
@@ -107,6 +113,21 @@ func (b *memoryBlob) EqualContent(ctx context.Context, other Blob) (_ bool, retE
...
+	}
+	return &modulev1alpha1.Blob{
+		Hash: &modulev1alpha1.Hash{
+			Kind:   hashKind,
+			Digest: b.digest.Bytes(),
+		},
+		Content: b.content,
+	}, nil
+}
+
```
Comment (twilly) [inline private/pkg/manifest/module.go:130]: I'm a little weary about adding mappers for an internal type. It ties it to a representation it doesn't necessarily care about. Imagine what a "fileBlob" or a "GCPBlob" implementation would be; they all would look quite similar.

```go
func AsProtoBlob(ctx contentex.Context, blob Blob) (*modulev1alpha1.Blob, error) {
	file, err := blob.Open(ctx)
	if err != nil {
		return nil, err
	}
	content, err := io.ReadAll(file)
	if err != nil {
		return nil, err
	}
	blob := &modulev1alpha1.Blob{
		Hash: &modulev1alpha1.Hash{
			Kind:   hashKind,
			Digest: blob.Digest().Bytes(),
		},
		Content: content,
	}
	if err := file.Close(); err != nil {
		return nil, err
	}
	return blob, nil
}
```

Comment (unmultimedio) [reply private/pkg/manifest/module.go]: thx! done

**Review by twilly — APPROVED** [review]
:shipit:

**Pull request merged**